db-tests/src/test/java/gov/nasa/jpl/aerie/database/ExternalEventTests.java

```diff
@@ -0,0 +1,1922 @@
+    ExternalEvent twoC = new ExternalEvent("2", "DerivationB", "Derivation_Test_02.json", dg, "2024-01-09 11:00:00+00", "01:05:00", "{\"notes\": \"replaces 2 in test 01, despite different event type\", \"rules\": [4], \"should_present\": true}");
+
+    ExternalEvent nine = new ExternalEvent("9", "DerivationC", "Derivation_Test_03.json", dg, "2024-01-02 00:00:00+00", "01:00:00", "{\"notes\": \"not subsumed\", \"rules\": [1], \"should_present\": true}");
+
+    // Third, insert types
+    String[] externalEventTypes = {"DerivationA", "DerivationB", "DerivationC", "DerivationD"};
+    for (String eventType : externalEventTypes) {
+      insertExternalEventType(eventType);
+    }
+    insertExternalSourceType("Test");
+    insertDerivationGroup(dg, "Test");
+
+    // Then, insert sources
+    insertExternalSource(sourceOne);
+    insertExternalSource(sourceTwo);
+    insertExternalSource(sourceThree);
+    insertExternalSource(sourceFour);
+
+    // Finally, insert events
+    insertExternalEvent(twoA);
+    insertExternalEvent(seven);
+    insertExternalEvent(eight);
+    insertExternalEvent(one);
+    insertExternalEvent(twoB);
+    insertExternalEvent(three);
+    insertExternalEvent(four);
+    insertExternalEvent(five);
+    insertExternalEvent(six);
+    insertExternalEvent(twoC);
+    insertExternalEvent(nine);
+  }
+
+  void insertStandardTypes() throws SQLException {
+    // insert external event type
+    insertExternalEventType(et);
+
+    // insert external source type
+    insertExternalSourceType(st);
+
+    // insert derivation group
+    insertDerivationGroup(dg, st);
+  }
+  //endregion
+
+
+  //region Basic Tests
+  /**
+   * A set of basic tests that verify the general functionality of uploading sources, derived events, the
+   *    derivation_group_comp view, and deletion.
+   * Aside from the derivation_group_comp view test, these tests are not rigorous (more rigorous tests follow in the
+   *    ConstraintsTests class).
+   */
+  @Nested
+  @TestInstance(TestInstance.Lifecycle.PER_CLASS)
+  class BasicTests {
+    /**
+     * This first test is to verify that a simple upload works, regardless of if its output is correct.
+     * As we do not upload a file we just run the queries that would be run if files were uploaded,
+     * namely tests 01-04 from SSMO-MPS/mission-data-sandbox/derivation_test.
+     */
+    @Test
+    void uploadWithoutError() {
+      assertDoesNotThrow(() -> upload_source(dg));
+    }
+
+
+    /**
+     * This test uploads data (tests 01-04 from SSMO-MPS/mission-data-sandbox/derivation_test) and then confirms derived
+     * events associated with the upload perform as expected. All desired keys should be present.
+     * This is an extension of uploadWithoutError.
+     */
+    @Test
+    void basicDerivedEvents() throws SQLException {
```

Comment:
This test is not providing any value as it is testing your helper method and not actually production code. Please remove it.

A test that _would_ provide value would be one that validates that an external source JSON is extracted and uploaded correctly, but such a test would belong in the UI PR and not the backend PR.